weed/storage/volume.go

```diff
@@ -25,6 +25,8 @@ type Volume struct {
 	Collection         string
 	DataBackend        backend.BackendStorageFile
 	nm                 NeedleMapper
+	tmpNm              *NeedleMap
```

Comment:
why one NeedleMapper is enough for both cases, but the temp needle mapper needs to have different implementation?

Reply:
NeedleMapper is a interface.  tmpNm and tmpLnm are point of NeedleMap,  they have to be modified to store necessary information when vacuuming and committing volume, and make NeedleMapper(v.nm) to piont to them at last.
If I was wrong or this solution could be optimized, please let me know.I'll refactor it.

Reply:
Done